src/iso19111/io.cpp

```diff
@@ -4072,7 +4072,7 @@ WKTParser::Private::buildProjectedCRS(const WKTNodeNNPtr &node) {
             props.set(IdentifiedObject::NAME_KEY, "WGS 84 / UPS North (E,N)");
         } else if (projCRSName == "UPS_South") {
             props.set(IdentifiedObject::NAME_KEY, "WGS 84 / UPS South (E,N)");
-        } else {
+        } else if (cartesianCS) {
```

Comment:
Do you still need a final else clause to catch things or error?

Reply:
the lack of cartesianCS will be eventually be caught later in the method at line 4212